jobs/build.Jenkinsfile

```diff
@@ -442,50 +378,22 @@ lock(resource: "build-${params.STREAM}") {
-                """)
-                try {
-                    parallelruns = [:]
-                    parallelruns['metal'] = {
-                        shwrap("cosa kola testiso -S --output-dir tmp/kola-testiso-metal")
-                    }
-                    parallelruns['metal4k+uefi'] = {
-                        shwrap("cosa kola testiso -SP --qemu-native-4k --qemu-multipath --output-dir tmp/kola-testiso-metal4k")
-                        shwrap("cosa shell -- mkdir -p tmp/kola-testiso-uefi")
-                        shwrap("cosa kola testiso -S --qemu-firmware=uefi --scenarios iso-live-login,iso-as-disk --output-dir tmp/kola-testiso-uefi/insecure")
-                        shwrap("cosa kola testiso -S --qemu-firmware=uefi-secure --scenarios iso-live-login,iso-as-disk --output-dir tmp/kola-testiso-uefi/secure")
-                    }
-                    // process this batch
-                    parallel parallelruns
-                } catch (Throwable e) {
-                    throw e
-                } finally {
-                    shwrap("""
-                    cosa shell -- tar -c --xz tmp/kola-testiso-metal/ > kola-testiso-metal.tar.xz
-                    cosa shell -- tar -c --xz tmp/kola-testiso-metal4k/ > kola-testiso-metal4k.tar.xz
-                    cosa shell -- tar -c --xz tmp/kola-testiso-uefi/ > kola-testiso-uefi.tar.xz
-                    """)
-                    archiveArtifacts allowEmptyArchive: true, artifacts: 'kola-testiso*.tar.xz'
-
-                    // For now we want to notify ourselves when this workaround is observed. It won't
-                    // fail the build, just give us information.
-                    // https://github.com/coreos/fedora-coreos-tracker/issues/1233
-                    def grepRc = shwrapRc("""
-                         cosa shell -- grep 'tracker issue workaround engaged for .*issues/1233' \
-                            tmp/kola-testiso-uefi/insecure/{iso-live-login,iso-as-disk}/console.txt
-                    """)
-                    if (grepRc == 0) {
-                        warnError(message: 'Detected used workaround for #1233') {
-                            error('Detected used workaround for #1233')
-                        }
-                    }
+            // Run Kola TestISO tests for metal artifacts
+            kolaTestIso(cosaDir: env.WORKSPACE, arch: basearch)
+            // For now we want to notify ourselves when a particular workaround is observed.
+            // It won't fail the build, just give us information.
+            // https://github.com/coreos/fedora-coreos-tracker/issues/1233
+            // XXX: This relies on implementation details in kolatestIso(),
+            //      but since this is a hack and probably short lived that's OK.
+            // First check to make sure the files exist, then grep for the workaround.
+            shwrap("ls tmp/kolaTestIso-*/kola-testiso-uefi/insecure/{iso-live-login,iso-as-disk}/console.txt")
```

Comment:
Shouldn't this be prefixed by `cosa shell -- `?

Reply:
yes and no :)